### PR TITLE
Pref grid

### PR DESCRIPTION
**Pull request by akersinformz**
Introduce NgZones into Data Grid to help manage scrolling and click events

### CODE REVIEW

File: src/data-grid/data-grid.component.html [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
     </div>
-    <div class="js-innerPs" [ngStyle]="{'min-width': widget.contentWidth, 'height': 'calc(' + widget.containerHeight + ' - ' + header.offsetHeight + 'px)'}" [suprePerfectScrollbar]="widget.ps || {}" (ps-scroll-x)="onPsXScroll($event)">
+    <!--header starts-->
+    <!--body starts-->
+    <div class="js-innerPs"  supreStickyContainerHeader  [ngStyle]="dgStyle" >
       <div class="ps-content">
         <ng-content select="supre-data-grid-body"></ng-content>
       </div>
     </div>
+    <!--body starts-->
```
Comment (xerona) [inline src/data-grid/data-grid.component.html]: not a big deal, but you have `<!--header starts-->` and `<!--body starts-->` twice.

File: src/data-grid/data-grid.component.scss [hunk 2]
```diff
@@ -90,3 +88,30 @@ supre-data-grid-column[type="checkbox"] {
...
+//
+// NOTE: Removed/Changed the styles in audri for hiding the scroll bar
+.js-innerPs {
+  overflow-y: scroll;
+  overflow-x: hidden;
+  &::-webkit-scrollbar-thumb {
+    background-color: transparent;
+  }
+  &::-webkit-scrollbar-track {
+    background-color: #e8ecf0;
```
Comment (xerona) [inline src/data-grid/data-grid.component.scss]: `$ice`

File: src/data-grid/data-grid.component.scss [hunk 3]
```diff
@@ -90,3 +88,30 @@ supre-data-grid-column[type="checkbox"] {
...
+  }
+  &::-webkit-scrollbar {
+    width: 12px;
+    background-color: transparent;
+  }
+  &::-webkit-scrollbar-thumb {
+    border-radius: 10px;
+    -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,.3);
+    background-color: #55a0c2;//#555;
```
Comment (xerona) [inline src/data-grid/data-grid.component.scss]: `$aqua`

File: src/data-grid/data-grid.component.ts [hunk 4]
```diff
@@ -56,40 +79,7 @@ export class DataGridComponent implements OnChanges, OnInit, AfterViewInit, OnDe
...
-        .mapTo(0)
-        .subscribe(left => this.setColumns(left));
-
-      this.stateManagerService.getModel
-        .map(model => model.isColumnLocked)
-        .filter(isColumnLocked => isColumnLocked != null)
-        .distinctUntilChanged()
-        .filter(isColumnLocked => !!isColumnLocked)
-        .mapTo(document.querySelector('.js-outerPs'))
-        .subscribe(outerPs => this.setAllYScroll(outerPs));
```
Comment (craigmichaelmartin) [inline src/data-grid/data-grid.component.ts]: Wow, I can't believe I wasn't unsubscribing from these. What a bad job out of me! Thanks for cleaning this up, amidst your other perf optimizations!!

File: src/data-grid/data-grid.component.scss (hunk 3 above)
Comment (akersinformz) [inline src/data-grid/data-grid.component.scss]: All set

File: src/data-grid/data-grid.component.scss (hunk 2 above)
Comment (akersinformz) [inline src/data-grid/data-grid.component.scss]: All set

File: src/data-grid/data-grid.component.html (hunk 1 above)
Comment (akersinformz) [inline src/data-grid/data-grid.component.html]: All set

**Comment by akersinformz** [comment]
**TODO:**
The Child row functionality will need to be reviewed again once needed.
Very slow to generate the rows.

**Pull request merged**